### DISCUSSION TITLE
TypeConverter is not compiling

### DISCUSSION THREAD

**Discussion by Enhed** [simolus3/drift#1757] [Q&A] [answered]
what wrong with my json converter? when i try run build runner got error: Too many elements

```dart
class JsonConverter extends TypeConverter<Map<String, dynamic>, String> {
  const JsonConverter();

  @override
  Map<String, dynamic>? mapToDart(String? fromDb) => fromDb == null ? null : jsonDecode(fromDb);

  @override
  String? mapToSql(Map<String, dynamic>? value) => value == null ? null : jsonEncode(value);
}

class Tests extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get value => text().map(const JsonConverter())();
}
```

**Comment by simolus3** [comment, marked as answer]
Thanks for the report! This has been fixed in `drift_dev` version 1.5.2.

@westito The problem was caused by the the runtime type (`Map<String, dynamic>`) having more than one type parameter because we have a `type.typeArguments.single.isDynamic` in the logic to determine whether we emit the source type in the generated code. I've changed that logic to be more general now and added a bunch of tests here: https://github.com/simolus3/drift/commit/5a2b6a36cc7be1530066d9eacd61b4cec7d667cc#diff-437dbbdba04bdac07788f041d92249b9684d0593bfbbb032f832a2ec4b19c117 Does that still cover your intention of generating the source type if it could refer to an unknown class? I want to avoid doing that if the type is explicitly written out as `dynamic`.

**Reply by westito** [reply]
Mine was just paying attention to whether the first type argument was dynamic or not. But now it can have more dynamic elements and parser can distinguish whether the user is using it intentionally or not. Perfect.
